### PR TITLE
feat(connector-besu): add getBlock web service #1065

### LINKED ISSUE #1065: feat(connector-besu): add getBlock web service
### Description
As a ... I want to ... so that ....

(add more description and relevant links)

### Acceptance Criteria
1. First acceptance criteria
2. ...
3. ...

### PR DESCRIPTION
**Pull request by twiggins120**
fixes #1065
Signed-off-by: Tommesha Wiggins <tommesha.wiggins@accenture.com>

### CODE REVIEW

**Review by ty-lazar — COMMENTED** [review]
@petermetz A bunch of files were deleted in this PR - are they supposed to be?

File: packages/cactus-plugin-ledger-connector-besu/src/main/typescript/plugin-ledger-connector-besu.ts [hunk 1]
```diff
@@ -200,7 +202,7 @@ export class PluginLedgerConnectorBesu
       endpoints.push(endpoint);
     }
     {
-      const endpoint = new RunTransactionEndpoint({
+      const endpoint = new GetBlockEndpoint({
```
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-besu/src/main/typescript/plugin-ledger-connector-besu.ts:205]: @twiggins120: Oops, looks like here you deleted the RunTransactionEndpoint , if you add that back then it should be good

cc: @ty-lazar

File: .cspell.json [hunk 2]
```diff
@@ -84,5 +84,11 @@
     "ignorePaths": [
         "node_modules/**",
         "**/build/**"
+    ],
```
Comment (petermetz) [inline .cspell.json:87]: @twiggins120 Does the linter fail without these?

cc: @ty-lazar

File: packages/cactus-test-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-validator-besu/get-block-endpoint.test.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  const configuration = new BesuApiClientOptions({ basePath: node1Host });
+  const api = new BesuApiClient(configuration);
+
+  // Test for 200 valid response test case
+  const res = await api.getBlockV1(request);
+
+  const { status, data } = res;
+  t.true(status >= 200, "status GTE 200 OK");
+  t.true(status < 300, "status LT 300 OK");
+  t.ok(data, "GetBlockResponse Truthy OK");
```
Comment (petermetz) [inline packages/cactus-test-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-validator-besu/get-block-endpoint.test.ts:139]: @twiggins120 Please also check if the data field has the expected contents (an object with the block information should be in there)

cc: @ty-lazar

**Review by petermetz — CHANGES_REQUESTED** [review]

**Comment by petermetz** [comment]
> @petermetz A bunch of files were deleted in this PR - are they supposed to be?

@ty-lazar Nope, those were just dropped because of the failing build, we took care of it in today's session, cheers!

File: packages/cactus-test-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-validator-besu/get-block-endpoint.test.ts (hunk 3 above)
Comment (twiggins120) [reply packages/cactus-test-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-validator-besu/get-block-endpoint.test.ts:139]: @petermetz  All of the requested changes have been resolved in the morning checkup!

**Review by petermetz — APPROVED** [review]

**Review by takeutak — APPROVED** [review]

File: packages/cactus-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-ledger-connector-besu/deploy-contract/get-block.test.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  });
+  const connector: PluginLedgerConnectorBesu = await factory.create({
+    rpcApiHttpHost,
+    rpcApiWsHost,
+    instanceId: uuidv4(),
+    pluginRegistry: new PluginRegistry({ plugins: [keychainPlugin] }),
+  });
+
+  const request: GetBlockV1Request = { blockHashOrBlockNumber: 0 };
+  const currentBlock = await connector.getBlock(request.blockHashOrBlockNumber);
```
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-ledger-connector-besu/deploy-contract/get-block.test.ts:57]: @twiggins120 

```suggestion
  const currentBlock = await connector.getBlock(request);
```

**Review by petermetz — CHANGES_REQUESTED** [review]

File: packages/cactus-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-ledger-connector-besu/deploy-contract/get-block.test.ts (hunk 4 above)
Comment (petermetz) [reply packages/cactus-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-ledger-connector-besu/deploy-contract/get-block.test.ts:57]: @twiggins120 @ty-lazar I see this one got marked as resolved, but the code appears to be unchanged?

**Review by petermetz — CHANGES_REQUESTED** [review]

File: packages/cactus-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-ledger-connector-besu/deploy-contract/get-block.test.ts (hunk 4 above)
Comment (ty-lazar) [reply packages/cactus-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-ledger-connector-besu/deploy-contract/get-block.test.ts]: @twiggins120 @petermetz It appears to be resolved now.

**Review by petermetz — APPROVED** [review]

**Comment by petermetz** [comment]
@twiggins120 Could you double check if everything was committed? The CI says[1] that the code generator has changed the api.ts file which would indicate that we forgot to commit something yesterday.

https://github.com/hyperledger/cactus/pull/1090/checks?check_run_id=3022920496#step:4:50229

**Comment by twiggins120** [comment]
@petermetz I checked in VS and there is nothing else to commit.

**Pull request merged**